python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -184,6 +187,34 @@ def run_iteration(
         self._logger.info("Finished auto-materialization tick")
 
 
+def get_external_job_name_for_asset_keys(
+    asset_keys: Sequence[AssetKey], asset_graph: AssetGraph, external_repository
+) -> Optional[str]:
+    # for observable source assets, require a hack
+    if all(asset_graph.is_observable(asset_key) for asset_key in asset_keys):
+        target_partitions_def = asset_graph.get_partitions_def(asset_keys[0])
+
+        partitions_def_by_job_name = {}
+        for (
+            external_partition_set_data
+        ) in external_repository.external_repository_data.external_partition_set_datas:
+            if external_partition_set_data.external_partitions_data is None:
+                continue
+            job_name = external_partition_set_data.job_name
+            partitions_def = (
+                external_partition_set_data.external_partitions_data.get_partitions_definition()
+            )
+            partitions_def_by_job_name[job_name] = partitions_def
+        for job_name, external_partitions_def in partitions_def_by_job_name.items():
+            if not job_name.startswith(ASSET_BASE_JOB_PREFIX):
+                continue
+            if external_partitions_def == target_partitions_def:
+                return job_name
+        return None
+    else:
+        return asset_graph.get_implicit_job_name_for_assets(asset_keys)
```

Comment:
Is it possible to just modify `get_implicit_job_name_for_assets` with this logic?

Reply:
I wanted to avoid having to pass in the external repository object into that function, as that's pretty hard to get access to. another approach would be to make that object optional, or to just pass in mock objects into the existing tests -- thoughts?

Reply:
eh on second thought, it's not hard to get access to -- updating